### PR TITLE
Added IntermediateOutputPath option to MonoAotCompiler task parameters

### PR DESCRIPTION
**Pull request by naricc**
This adds the `IntermediateOutputPath` parameter to the MonoAotCompiler task in generated projects that use the MonoAotLLVM toolchain. That parameter is now required.

fixes: https://github.com/dotnet/BenchmarkDotNet/issues/1887

### CODE REVIEW

**Comment by naricc** [comment]
@adamsitnik

**Review by adamsitnik — CHANGES_REQUESTED** [review]
Please delete the `.BenchmarkDotNet.Samples.csproj.swp` file (for some reason I can't do it from GH anymore)

**Comment by naricc** [comment]
> Please delete the `.BenchmarkDotNet.Samples.csproj.swp` file (for some reason I can't do it from GH anymore)

Done.

**Comment by adamsitnik** [comment]
@naricc You have removed `BenchmarkDotNet.Samples.csproj.swp`, but added `samples/BenchmarkDotNet.Samples/.out.swp` and `samples/BenchmarkDotNet.Samples/out`. Could you please remove all temporary files?

File: src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMBuilder.cs [hunk 1]
```diff
@@ -21,48 +21,8 @@ public BuildResult Build(GenerateResult generateResult, BuildPartition buildPart
...
-            }
-
-            foreach (string fileName in publishDirFiles)
-            {
-                string newFileName = Path.ChangeExtension(fileName, sharedObjectExtension);
-
-                logger.WriteLine($"RenameSharedLibraries: Moving ${fileName} to {newFileName}");
-                File.Move(fileName, newFileName);
-            }
-        }
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMBuilder.cs:66]: after the removal of all the custom logic, this type has became just a dummy wrapper for `DotNetCliBuilder`. Could you please remove it and use `DotNetCliBuilder` directly for the AOT toolchain?

https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMToolChain.cs#L20-L22

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @naricc !

**Pull request merged**